front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -259,17 +261,41 @@ export default function AssistantBuilder({
   const defaultScope =
     flow === "workspace_assistants" ? "workspace" : "private";
 
-  const [builderState, setBuilderState] = useState<AssistantBuilderState>({
-    ...DEFAULT_ASSISTANT_STATE,
-    scope: initialBuilderState?.scope ?? defaultScope,
-    generationSettings: {
-      ...DEFAULT_ASSISTANT_STATE.generationSettings,
-      modelSettings:
-        plan.code === FREE_TEST_PLAN_CODE
-          ? GPT_3_5_TURBO_MODEL_CONFIG
-          : GPT_4_TURBO_MODEL_CONFIG,
-    },
-  });
+  const [builderState, setBuilderState] = useState<AssistantBuilderState>(
```

Comment:
Context: we want to inject the props `initialBuilderState` only once, after that the state is fully controller by the component, and not by the props.

Reply:
Using useEffect() to control that was leading to a race condition between the setting of the initial state and the setting of a random avatar.